File: preview/logical/python/mlir/dialects/phys.py
# ============================================================================ #
# Copyright (c) 2026 NVIDIA Corporation & Affiliates.                          #
# All rights reserved.                                                         #
#                                                                              #
# This source code and the accompanying materials are made available under     #
# the terms of the Apache License 2.0 which accompanies this distribution.     #
# ============================================================================ #
"""Import sentinel for the native-only ``phys`` MLIR dialect.

The dialect has no generated Python OpViews or type casters. Keeping this
module importable lets MLIR cache that fact instead of retrying a failed import
for every generic operation, type, and SSA value downcast.
"""

__all__ = ()

File: preview/logical/ir/lib/CAPI/Dialect/Phys.cpp
/*******************************************************************************
 * Copyright (c) 2026 NVIDIA Corporation & Affiliates.                         *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 *******************************************************************************/

#include "qlx-c/Dialect/Phys.h"
#include "qlx/Dialect/Phys/IR/PhysDialect.h"
#include "mlir/CAPI/Registration.h"
MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(Phys, phys, qlx::phys::PhysDialect)
